cmd/login/runner_test.go

```diff
@@ -114,6 +118,103 @@ func TestLogin(t *testing.T) {
 	}
 }
 
+func TestMCLoginWithInstallation(t *testing.T) {
+	testCases := []struct {
+		name        string
+		startConfig *clientcmdapi.Config
+		flags       *flag
+		expectError *microerror.Error
+	}{
+		// empty start config
+		{
+			name:        "case 0",
+			startConfig: &clientcmdapi.Config{},
+			flags:       &flag{},
+		},
+		// filled start config
+		{
+			name:        "case 1",
+			startConfig: createValidTestConfigMC(),
+			flags:       &flag{},
+		},
+		// self contained file
+		{
+			name:        "case 2",
+			startConfig: createValidTestConfigMC(),
+			flags: &flag{
+				SelfContained: "/codename.yaml",
+			},
+		},
+		// keeping WC context
+		{
+			name:        "case 3",
+			startConfig: createValidTestConfigWC(),
+			flags: &flag{
+				KeepContext: true,
+			},
+		},
+	}
+
+	for i, tc := range testCases {
+		t.Run(strconv.Itoa(i), func(t *testing.T) {
+			configDir, err := os.MkdirTemp("", "loginTest")
+			if err != nil {
+				t.Fatal(err)
+			}
+			fs := afero.NewOsFs()
+			if len(tc.flags.SelfContained) > 0 {
+				tc.flags.SelfContained = configDir + tc.flags.SelfContained
+			}
+
+			r := runner{
+				k8sConfigAccess: &clientcmd.ClientConfigLoadingRules{
+					ExplicitPath: configDir + "/config.yaml",
+				},
+				flag:   tc.flags,
+				stdout: new(bytes.Buffer),
+				fs:     afero.NewBasePathFs(fs, configDir),
+			}
+			err = clientcmd.ModifyConfig(r.k8sConfigAccess, *tc.startConfig, false)
+			if err != nil {
+				t.Fatal(err)
+			}
+			originConfig, err := r.k8sConfigAccess.GetStartingConfig()
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			ctx := context.TODO()
+			r.setLoginOptions(ctx, []string{"codename"})
+
+			err = r.loginWithInstallation(ctx, "token", CreateTestInstallation())
```

Comment:
We add the token override here to avoid the OIDC process. For that we need a separate test.